dbms/src/AggregateFunctions/AggregateFunctionWindowFunnel.h

```diff
@@ -137,34 +135,37 @@ struct AggregateFunctionWindowFunnelData
   * Usage:
   * - windowFunnel(window)(timestamp, cond1, cond2, cond3, ....)
   */
+template <typename T, typename Data>
 class AggregateFunctionWindowFunnel final
-    : public IAggregateFunctionDataHelper<AggregateFunctionWindowFunnelData, AggregateFunctionWindowFunnel>
+    : public IAggregateFunctionDataHelper<Data, AggregateFunctionWindowFunnel<T, Data>>
 {
 private:
-    UInt32 window;
+    UInt64 window;
     UInt8 events_size;
 
 
     // Loop through the entire events_list, update the event timestamp value
     // The level path must be 1---2---3---...---check_events_size, find the max event level that statisfied the path in the sliding window.
     // If found, returns the max event level, else return 0.
     // The Algorithm complexity is O(n).
-    UInt8 getEventLevel(const AggregateFunctionWindowFunnelData & data) const
+    UInt8 getEventLevel(const Data & data) const
     {
         if (data.size() == 0)
             return 0;
         if (events_size == 1)
             return 1;
 
-        const_cast<AggregateFunctionWindowFunnelData &>(data).sort();
+        const_cast<Data &>(data).sort();
 
-        // events_timestamp stores the timestamp that latest i-th level event happen withing time window after previous level event.
-        // timestamp defaults to -1, which unsigned timestamp value never meet
-        std::vector<Int32> events_timestamp(events_size, -1);
+        /// events_timestamp stores the timestamp that latest i-th level event happen withing time window after previous level event.
+        /// timestamp defaults to -1, which unsigned timestamp value never meet
+        /// there may be some bugs when UInt64 type timstamp overflows Int64, but it works on most cases.
```

Comment:
@alexey-milovidov  should this limitation be noted at docs?   `timestamp + window`  can't be overflow maximum of Int64

Reply:
Sure.